src/poetry/utils/env.py

```diff
@@ -1145,7 +1145,7 @@ def get_base_prefix(cls) -> Path:
     def generate_env_name(cls, name: str, cwd: str) -> str:
         name = name.lower()
         sanitized_name = re.sub(r'[ $`!*@"\\\r\n\t]', "_", name)[:42]
-        normalized_cwd = os.path.normcase(cwd)
+        normalized_cwd = os.path.realpath(os.path.normcase(cwd))
```

Comment:
```suggestion
        normalized_cwd = os.path.normcase(os.path.realpath(cwd))
```

It should be the other way around to ensure that case is normalized.

Reply:
Good catch, thanks.

I suspect it's a bit redundant, because `realpath()` does seem to return the canonical case for the path (on NTFS at least), with an uppercase drive letter, so it should already be consistent. Calling `normcase()` last should keep the majority of environment names unchanged by this PR,  though, so worth doing.